File: coinsaver-api/src/main/java/com/coinsaver/api/dtos/request/UpdateTransactionRequestDto.java
package com.coinsaver.api.dtos.request;

import com.coinsaver.core.enums.StatusType;
import com.coinsaver.core.enums.TransactionCategoryType;
import com.coinsaver.core.enums.TransactionType;
import com.coinsaver.core.enums.UpdateTransactionType;
import com.coinsaver.domain.entities.FixTransaction;
import com.coinsaver.domain.entities.InstallmentTransaction;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.modelmapper.ModelMapper;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class UpdateTransactionRequestDto {

    @NotNull
    private BigDecimal amount;
    private LocalDate payDay;
    @NotBlank
    private String description;
    @NotNull
    private StatusType status;
    @NotNull
    private TransactionCategoryType category;
    @NotNull
    private Boolean fixedExpense;
    private Integer repeat;
    @NotNull
    private UpdateTransactionType updateTransactionType;
    @NotNull
    private TransactionType transactionType;
    @NotNull
    private Long transactionId;
    private Long installmentTransactionId;
    private Long fixTransactionId;

    public FixTransaction convertDtoToFixTransactionEntity() {
        return new ModelMapper().map(this, FixTransaction.class);
    }
}

File: coinsaver-front/src/app/dtos/transactions/request/update-transaction.request.dto.ts
import { TransactionCategoryTypeEnum } from 'src/app/enums/transaction-category-type.enum';
import { StatusTypeEnum } from '../../../enums/status-type.enum';
import { TransactionTypeEnum } from 'src/app/enums/transaction-type.enum';
import { UpdateTransactionTypeEnum } from 'src/app/enums/update-transaction-type.enum';

export class UpdateTransactionRequestDto {
  constructor(
    public amount?: number,

    public payDay?: string | Date,


    public description?: string,

    public status?: StatusTypeEnum,
    public category?: TransactionCategoryTypeEnum,

    public fixedExpense?: boolean,

    public repeat?: number,

    public updateTransactionType?: UpdateTransactionTypeEnum,
    public transactionType?: TransactionTypeEnum,

    public transactionId?: number,
    public installmentTransactionId?: number,
    public fixTransactionId?: number,
  ) {}
}
